connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/DeadLetterQueueReporter.java

```diff
@@ -121,20 +121,22 @@ public static DeadLetterQueueReporter createAndSetup(Map<String, Object> adminPr
     /**
      * Write the raw records into a Kafka topic and return the producer future.
      *
-     * @param context processing context containing the raw record at {@link ProcessingContext#consumerRecord()}.
+     * @param context processing context containing the raw record at {@link ProcessingContext#original()}.
      * @return the future associated with the writing of this record; never null
      */
-    public Future<RecordMetadata> report(ProcessingContext context) {
+    @SuppressWarnings("unchecked")
+    public Future<RecordMetadata> report(ProcessingContext<?> context) {
         if (dlqTopicName.isEmpty()) {
             return CompletableFuture.completedFuture(null);
         }
         errorHandlingMetrics.recordDeadLetterQueueProduceRequest();
 
-        ConsumerRecord<byte[], byte[]> originalMessage = context.consumerRecord();
-        if (originalMessage == null) {
+        if (!(context.original() instanceof ConsumerRecord)) {
```

Comment:
I wish we could retain full type safety here but if it's too much work to be worth it then we can live with this. Do you think we can at least log a warning inside this branch if `context.original()` is non-null, though?

Reply:
This is another generic-oddity that I fixed with a follow-up refactor that makes the DeadLetterQueueReporter#report accept a ProcessingContext<ConsumerRecord<byte[], byte[]>> and eliminates all of these runtime checks which don't ever actually fail.

Reply:
Ah, nice. I'll mark this resolved and look forward to the refactor 👍